Additional GTG output "CIT"

hsinmulin-NOAA opened the pull request:
Related changes for GTG new output "CIT" in the source code & model input files.
The new "CIT"  (CITEDR in GRID2 Table) 
("0  19    50   0  CITEDR" as in https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table4-2-0-19.shtml) 

In additional to the added new variable, the consistency of UPP outputs with or without variables from GTG are considered.     
The new files will be committed according to this update are 

/sorc/ncep_post.fd
MISCLN.f
MDLFLD.f
gtg_interp.F90 : Stub code for GTG protection but to make UPP public to work
"CMakeLists.txt" to include new file "gtg_interp.F90" from GTG.  https://github.com/NOAA-EMC/UPP/issues/840

/parm
fv3lam_rrfs.xml
params_grib2_tbl_new.text
params_grib2_tbl_new
post_avblflds.xml
postxconfig-NT-fv3lam_rrfs.txt

New GTG_4.15 version link (protected)

hsinmulin-NOAA committed:
add CIT 

add "CIT" as additional GTG output

hsinmulin-NOAA committed:
add CIT

additional "CIT" related GTG output changes for the UPP

hsinmulin-NOAA committed:
add CIT

additional GTG output variable "CIT"

FernandoAndrade-NOAA commented:
Hi @hsinmulin-NOAA, please update the changelog entries at the top of MISCLN.f for your updates and create/link a git issue for this pr for tracking. Thank you

WenMeng-NOAA commented on parm/params_grib2_tbl_new.text:
```diff
@@ -751,6 +751,8 @@
 !  Added more parameters in 12/03/2019
    0  19    37   0 ICESEV
+!  Added more parameters in 11/01/2023
+   0  19    50   0 CITEDR
```
@hsinmulin-NOAA There is no this variable in the table 4.2-0-19. You may send a request to @AndrewBenjamin-NOAA for the table update.

AndrewBenjamin-NOAA replied:
@WenMeng-NOAA @hsinmulin-NOAA 

I will add CITEDR to the g2tmpl and include it in the PR I have open with them. I will provide the sample file when finished. 

As a reminder, CITEDR does not appear in v32 of the WMO grib2 update that released 11/30. However, it was accepted and approved for May 2024. Since it was already approved, I will add it to the g2tmpl, it just won't appear in the WMO grib2 documentation until May 2024.

WenMeng-NOAA replied:
@AndrewBenjamin-NOAA Thanks for updating g2tmpl. Can you also update the table at https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table4-2-0-19.shtml so RRFS developers can get reference of this new product?

AndrewBenjamin-NOAA replied:
Yes, I am doing that currently for the v32 release that just came out. I've been out the last few days so I haven't had a chance to complete every table, but I have added CITEDR to the list of products that need to be added to Table 4.2-0-19.

WenMeng-NOAA replied:
@AndrewBenjamin-NOAA When the updated params_grib2_tbl_new* files are available, please pass them to @hsinmulin-NOAA so he may include in this PR.

hsinmulin-NOAA commented:
The GitHub issue link is #820 <https://github.com/NOAA-EMC/UPP/issues/820>

The detail changelog about " MISCLN.f" can be found at
https://github.com/hsinmulin-NOAA/EMC_post/commit/3d47e0bf873b563a7ae0dad5c36b35ff740c3ade

On Mon, Dec 11, 2023 at 4:45 PM Fernando Andrade - NOAA <
***@***.***> wrote:

> Hi @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA>, please update the
> changelog entries at the top of MISCLN.f for your updates and create/link a
> git issue for this pr for tracking. Thank you
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/UPP/pull/839#issuecomment-1850938542>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCPJV3ULJ4KYXAYLXD3YI55F3AVCNFSM6AAAAABAQJVPR2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQNJQHEZTQNJUGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

YaliMao-NOAA commented:
@hsinmulin-NOAA Based on the latest GTG code delivery, may you please include the update of sorc/ncep_post.fd/CMakeLists.txt by adding gtg_interp.F90?

hsinmulin-NOAA commented:
Hi, Yali

Let me create an issue related to it and commit it later.

There is an interesting issue for  " CMakeLists.txt " to include it or not
that I have mentioned in the GTG update discussion.  I need to clear why it
exists before we get it done.

On Fri, Dec 15, 2023 at 10:50 AM YaliMao-NOAA ***@***.***>
wrote:

> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> Based on the latest
> GTG code delivery, may you please include the update of
> sorc/ncep_post.fd/CMakeLists.txt by adding gtg_interp.F90?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/UPP/pull/839#issuecomment-1858093939>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCN2BLW4ZY3R45K2MLDYJRWSPAVCNFSM6AAAAABAQJVPR2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQNJYGA4TGOJTHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

WenMeng-NOAA commented:
@hsinmulin-NOAA Please sync your branch with the latest UPP develop and rename your folk from EMC_post to UPP. You may follow the instructions at https://github.com/NOAA-EMC/UPP/discussions/372 to rename folk.

WenMeng-NOAA commented:
> @hsinmulin-NOAA Based on the latest GTG code delivery, may you please include the update of sorc/ncep_post.fd/CMakeLists.txt by adding gtg_interp.F90?

@YaliMao-NOAA and @hsinmulin-NOAA I do not see gtg_interp.F90 from the submodule post_gtg with commit @a982870.

hsinmulin-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

hsinmulin-NOAA committed:
Update CMakeLists.txt

Add "gtg_interp.F90" per updated GTG_4.15 version that pulls all the interpolation to this file.

hsinmulin-NOAA commented:
> > @hsinmulin-NOAA Based on the latest GTG code delivery, may you please include the update of sorc/ncep_post.fd/CMakeLists.txt by adding gtg_interp.F90?
> 
> @YaliMao-NOAA and @hsinmulin-NOAA I do not see gtg_interp.F90 from the submodule post_gtg with commit @a982870.

Extra line in "CMakeLists.txt" to include new file "gtg_interp.F90" from GTG was added.
Per issue https://github.com/NOAA-EMC/UPP/issues/840

hsinmulin-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

WenMeng-NOAA commented:
> > > @hsinmulin-NOAA Based on the latest GTG code delivery, may you please include the update of sorc/ncep_post.fd/CMakeLists.txt by adding gtg_interp.F90?
> > 
> > 
> > @YaliMao-NOAA and @hsinmulin-NOAA I do not see gtg_interp.F90 from the submodule post_gtg with commit @a982870.
> 
> Extra line in "CMakeLists.txt" to include new file "gtg_interp.F90" from GTG was added. Per issue #840

@hsinmulin-NOAA You need to update the revision of the submodule post_gtg.fd to include gtg_interp.F90. @YaliMao-NOAA Could you help Hsin-mu on this?

YaliMao-NOAA commented:
@hsinmulin-NOAA I am sending instructions of how to update GTG revision number.

hsinmulin-NOAA committed:
update gtg code revision

WenMeng-NOAA commented:
@hsinmulin-NOAA A sub version of gtg_interp.F90 is needed under sorc/ncep_post.fd. You may refer to an example:
https://github.com/NOAA-EMC/UPP/blob/develop/sorc/ncep_post.fd/gtg_mlmodel.F90

hsinmulin-NOAA commented:
The version of the GTG has been updated per instructions

On Thu, Dec 21, 2023 at 4:22 PM YaliMao-NOAA ***@***.***>
wrote:

> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> I am sending
> instructions of how to update GTG revision number.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/UPP/pull/839#issuecomment-1866944281>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCPUSL54X34GTTB5YTTYKSSAFAVCNFSM6AAAAABAQJVPR2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQNRWHE2DIMRYGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

hsinmulin-NOAA committed:
Create gtg_interp.F90

Stub code for GTG protection but to make UPP public to work

hsinmulin-NOAA commented:
Added the sub version per request.

On Thu, Dec 21, 2023 at 5:00 PM WenMeng-NOAA ***@***.***>
wrote:

> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> A sub version of
> gtg_interp.F90 is needed under sorc/ncep_post.fd. You may refer to an
> example:
>
> https://github.com/NOAA-EMC/UPP/blob/develop/sorc/ncep_post.fd/gtg_mlmodel.F90
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/UPP/pull/839#issuecomment-1866983454>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCL33ESYJ6X75W3RHQTYKSWRBAVCNFSM6AAAAABAQJVPR2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQNRWHE4DGNBVGQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

WenMeng-NOAA commented:
@hsinmulin-NOAA and @YaliMao-NOAA Now this PR is in a good shape for testing. Thanks for quick actions!

WenMeng-NOAA commented:
@hsinmulin-NOAA @YaliMao-NOAA From my UPP standalone test for RRFS GTG, there are to many printouts in runtime log (more than 60000 lines). Please work on limiting these debug information.

WenMeng-NOAA commented:
The UPP standalone testing with RRFS NA 3-km data was conducted on WCOSS2. With 192 tasks (4nodes, 48task/node), the runtime of RRFS product generation was changed as:
```
Without new gtg dataset: 399 seconds
With new gtg dataset: 667 seconds
```
@hsinmulin-NOAA @YaliMao-NOAA @HuiyaChuang-NOAA Please see my test results at /u/wen.meng/ptmp/post_rrfs_2023060614-gtg.

hsinmulin-NOAA committed:
Update MDLFLD.f for GTG special issue

The code change is to speed up and resolved the non-identical UPP outputs when compare  GTG on  & off (used or not used).

YaliMao-NOAA commented on sorc/ncep_post.fd/MDLFLD.f:
```diff
@@ -4205,7 +4205,7 @@ SUBROUTINE MDLFLD
 !
 ! COMPUTE NCAR GTG turbulence
-      IF(gtg_on) then
+      IF(gtg_on .and. (IGET(464) > 0 .or. IGET(467) > 0 .or. IGET(470) > 0)) then
```
With this one line of change, other paramset products than GTG will not trigger gtg_algo. The runtime of standalone RRFS UPP with GTG increases by 12-15% from with GTG off

WenMeng-NOAA commented:
Two tests were conducted with your latest commit and RRFS NA 3km model output on WCOSS2 dogwood:
-----------------------
test  including new dataset AVIATI*.tm00 (total four datasets): runtime is 476 seconds for 196 tasks
    test location: /u/wen.meng/ptmp/post_rrfs_2023060614-gtg
test without new dataset (total three datasets): runtime is  395 seconds
    test location: /u/wen.meng/ptmp/post_rrfs_2023060614-no-gtg
------------------------
@hsinmulin-NOAA and @YaliMao-NOAA Please verify AVIATI10.tm00 in my test /u/wen.meng/ptmp/post_rrfs_2023060614-gtg.

Right now, the runtime of new AVIAT (gtg) product generation looks fine to me. I have some concerns about introducing too many printouts from gtg code:
--------------------------- 
From my run directory /u/wen.meng/ptmp/post_rrfs_2023060614-gtg:
1) Additional 192 files thread_output*
2) Additional 40000 lines of printout in runtime log outpost_nems_2023060700
---------------------------
I see a lot of printout in gtg code, such as:
```
gtg_smoothseams.F90:      write(iprt, *) "enter getseams"
gtg_smoothseams.F90:      write(iprt, *) "seamfile=", seamfile
gtg_smoothseams.F90:      write(iprt, *) "npts=",npts
gtg_smoothseams.F90:       write(iprt,*) 'error finding free unit in getseams'
gtg_smoothseams.F90:       write(iprt,*) 'iunit=',iunit
gtg_smoothseams.F90:       write(iprt,*) 'ABORT'
gtg_smoothseams.F90:      write(iprt, *) 'Cannot open tile file ', trim(seamfile), ', iostat = ', iret
gtg_smoothseams.F90:          write(iprt, *) 'Error in reading tile line, iostat=', iret
gtg_smoothseams.F90:          write(iprt, *) 'Error in reading tile line, iostat=', iret
gtg_smoothseams.F90:          write(iprt, *) 'Error in getseams: array bounds exceeded: npts_in,npts=',npts_in,npts
gtg_smoothseams.F90:      write(iprt, *) "exit  getseams: iret=",iret
gtg_smoothseams.F90:      write(iprt, *) "fillnumber=", fillnumber
gtg_smoothseams.F90:      write(iprt, *) "There are", npts, "of imprinting grid points"
gtg_smoothseams.F90:       write(*,*) "The seam filling area exceeds UPP halo", fillnumber
map_routines.F90:        write(iprt,*)'enter get_map_factor: map_proj=',proj
map_routines.F90:        write(iprt,*)'nx,ny=',nx,ny
map_routines.F90:          write(iprt,*)'Getting mx,my for PROJ_LATLON: dlat,dxm,dym=',dlat,dxm,dym
map_routines.F90:          write(iprt,*)'Getting mx,my for PROJ_LC: truelat1,truelat2,cone=',truelat1,truelat2,cone
map_routines.F90:          write(iprt,*)'Getting mx,my for PROJ_PS: truelat1=',truelat1
map_routines.F90:          write(iprt,*)'Getting mx,my for PROJ_MERC: truelat1=',truelat1
map_routines.F90:          write(iprt,*)'Getting mx,my for PROJ_CYL: dlat,dxm,dym=',dlat,dxm,dym
map_routines.F90:           write(iprt,*)'Getting mx,my for PROJ_GAUSS: calling getmxmyglobal'
map_routines.F90:          write(iprt,*)'Error in get_map_factor: dxm undefined=',dxm
map_routines.F90:          write(iprt,*)'Getting mx,my for PROJ_ROTLL: use_rotll_comp_grid=',use_rotll_comp_grid
map_routines.F90:        write(iprt,*)'invalid input map projection in get_map_factor'
map_routines.F90:        write(iprt,*) 'exit  get_map_factor'
```
I would suggest to add a flag "debug_on" in gtg code to limit most of these debug information.

WenMeng-NOAA commented:
@hsinmulin-NOAA @YaliMao-NOAA For your reference, please see
https://github.com/NOAA-EMC/UPP/blob/develop/sorc/ncep_post.fd/INITPOST_NETCDF.f#L932
for debugprint option.

hsinmulin-NOAA committed:
Update MDLFLD.f

Add comments for the change log

YaliMao-NOAA reviewed (approved):
Output, run time and regression test look file. The PR can be approved. One issue has been opened #851 for too many printouts, which will be fixed in the next GTG delivery version.

YaliMao-NOAA commented:
> Please verify AVIATI10.tm00 in my test /u/wen.meng/ptmp/post_rrfs_2023060614-gtg

Found a bug.

It turns out the last vertical level of CIT in  AVIATI10.tm00 is marked wrongly, which should be 15240m, not -1.11992e+09m
204:798621407:d=2023060614:var discipline=0 master_table=2 parmcat=19 parm=50:14935 m above mean sea level:10 hour fcst:
205:801066854:d=2023060614:var discipline=0 master_table=2 parmcat=19 parm=50:-1.11992e+09 m above mean sea level:10 hour fcst:

As a comparison:
153:634892014:d=2023060614:MWTURB:14935 m above mean sea level:10 hour fcst:
154:637554115:d=2023060614:MWTURB:15240 m above mean sea level:10 hour fcst:

hsinmulin-NOAA committed:
Update MDLFLD.f

correct 470 to 477 to consistent with CITEDR in "post_avblflds.xml"

YaliMao-NOAA commented:
@hsinmulin-NOAA 

1. In MDLFLD.f, may you please change 477 back to 470? To run gtg_algo(), we only need to check EDPARM IDs to pull the trigger since EDPARM is kind of subset of CAT/MWT/CIT. I also added a line to the log history. Please compare my code on Cactus:
/lfs/h2/emc/vpppg/noscrub/yali.mao/git/UPP.v17_hsinmu/sorc/ncep_post.fd/MDLFLD.f

2. To fix the issue that the last vertical level of CIT in AVIATI10.tm00 of RRFS GTG is marked wrongly, I have a fix on Cactus:
/lfs/h2/emc/vpppg/noscrub/yali.mao/git/UPP.v17_hsinmu/sorc/ncep_post.fd/MISCLN.f
/lfs/h2/emc/vpppg/noscrub/yali.mao/git/UPP.v17_hsinmu/parm/fv3lam_rrfs.xml
/lfs/h2/emc/vpppg/noscrub/yali.mao/git/UPP.v17_hsinmu/parm/postxconfig-NT-fv3lam_rrfs.txt

3. Please add your comments of your code changes into the log history at the beginning of each Fortran code

4. Please merge UPP code updates. Other PRs were merged after your PR was submitted.

hsinmulin-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

hsinmulin-NOAA committed:
Add files via upload

1. update the comments
2.  fix the issue that the last vertical level of CIT in AVIATI10.tm00 of RRFS GTG per MISCLN.f

hsinmulin-NOAA committed:
Add files via upload

fix the issue that the last vertical level of CIT in AVIATI10.tm00 of RRFS GTG per change in MISCLN.f

hsinmulin-NOAA commented:
Modification per suggestion has been done and tested.


On Wed, Jan 10, 2024 at 1:41 PM YaliMao-NOAA ***@***.***>
wrote:

> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA>
>
>    1.
>
>    In MDLFLD.f, may you please change 477 back to 470? To run gtg_algo(),
>    we only need to check EDPARM IDs to pull the trigger since EDPARM is kind
>    of subset of CAT/MWT/CIT. I also added a line to the log history. Please
>    compare my code on Cactus:
>
>    /lfs/h2/emc/vpppg/noscrub/yali.mao/git/UPP.v17_hsinmu/sorc/ncep_post.fd/MDLFLD.f
>    2.
>
>    To fix the issue that the last vertical level of CIT in AVIATI10.tm00
>    of RRFS GTG is marked wrongly, I have a fix on Cactus:
>
>    /lfs/h2/emc/vpppg/noscrub/yali.mao/git/UPP.v17_hsinmu/sorc/ncep_post.fd/MISCLN.f
>
>    /lfs/h2/emc/vpppg/noscrub/yali.mao/git/UPP.v17_hsinmu/parm/fv3lam_rrfs.xml
>
>    /lfs/h2/emc/vpppg/noscrub/yali.mao/git/UPP.v17_hsinmu/parm/postxconfig-NT-fv3lam_rrfs.txt
>    3.
>
>    Please add your comments of your code changes into the log history at
>    the beginning of each Fortran code
>    4.
>
>    Please merge UPP code updates. Other PRs were merged after your PR was
>    submitted.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/UPP/pull/839#issuecomment-1885418340>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCMYQWQOSIVWXG5NXJ3YN3OFVAVCNFSM6AAAAABAQJVPR2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQOBVGQYTQMZUGA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

hsinmulin-NOAA committed:
Update MDLFLD.f

 switching GTG max to gtgx3 from gtgx2.

hsinmulin-NOAA committed:
Update MDLFLD.f

add comment for the change

HuiyaChuang-NOAA commented:
> @hsinmulin-NOAA
> 
> 1. In MDLFLD.f, may you please change 477 back to 470? To run gtg_algo(), we only need to check EDPARM IDs to pull the trigger since EDPARM is kind of subset of CAT/MWT/CIT. I also added a line to the log history. Please compare my code on Cactus:
>    /lfs/h2/emc/vpppg/noscrub/yali.mao/git/UPP.v17_hsinmu/sorc/ncep_post.fd/MDLFLD.f
> 2. To fix the issue that the last vertical level of CIT in AVIATI10.tm00 of RRFS GTG is marked wrongly, I have a fix on Cactus:
>    /lfs/h2/emc/vpppg/noscrub/yali.mao/git/UPP.v17_hsinmu/sorc/ncep_post.fd/MISCLN.f
>    /lfs/h2/emc/vpppg/noscrub/yali.mao/git/UPP.v17_hsinmu/parm/fv3lam_rrfs.xml
>    /lfs/h2/emc/vpppg/noscrub/yali.mao/git/UPP.v17_hsinmu/parm/postxconfig-NT-fv3lam_rrfs.txt
> 3. Please add your comments of your code changes into the log history at the beginning of each Fortran code
> 4. Please merge UPP code updates. Other PRs were merged after your PR was submitted.

@hsinmulin-NOAA make sure you pick up above fixes from Yali

hsinmulin-NOAA commented:
Fixed and done per suggestion.

On Wed, Jan 24, 2024 at 9:58 AM HuiyaChuang-NOAA ***@***.***>
wrote:

> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA>
>
>    1. In MDLFLD.f, may you please change 477 back to 470? To run
>    gtg_algo(), we only need to check EDPARM IDs to pull the trigger since
>    EDPARM is kind of subset of CAT/MWT/CIT. I also added a line to the log
>    history. Please compare my code on Cactus:
>
>    /lfs/h2/emc/vpppg/noscrub/yali.mao/git/UPP.v17_hsinmu/sorc/ncep_post.fd/MDLFLD.f
>    2. To fix the issue that the last vertical level of CIT in
>    AVIATI10.tm00 of RRFS GTG is marked wrongly, I have a fix on Cactus:
>
>    /lfs/h2/emc/vpppg/noscrub/yali.mao/git/UPP.v17_hsinmu/sorc/ncep_post.fd/MISCLN.f
>
>    /lfs/h2/emc/vpppg/noscrub/yali.mao/git/UPP.v17_hsinmu/parm/fv3lam_rrfs.xml
>
>    /lfs/h2/emc/vpppg/noscrub/yali.mao/git/UPP.v17_hsinmu/parm/postxconfig-NT-fv3lam_rrfs.txt
>    3. Please add your comments of your code changes into the log history
>    at the beginning of each Fortran code
>    4. Please merge UPP code updates. Other PRs were merged after your PR
>    was submitted.
>
> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> make sure you pick up
> above fixes from Yali
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/UPP/pull/839#issuecomment-1908303268>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCNBOILQUFIILO536CTYQEORVAVCNFSM6AAAAABAQJVPR2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMBYGMYDGMRWHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

HuiyaChuang-NOAA commented:
> Fixed and done per suggestion.
> […](#)
> On Wed, Jan 24, 2024 at 9:58 AM HuiyaChuang-NOAA ***@***.***> wrote: @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> 1. In MDLFLD.f, may you please change 477 back to 470? To run gtg_algo(), we only need to check EDPARM IDs to pull the trigger since EDPARM is kind of subset of CAT/MWT/CIT. I also added a line to the log history. Please compare my code on Cactus: /lfs/h2/emc/vpppg/noscrub/yali.mao/git/UPP.v17_hsinmu/sorc/ncep_post.fd/MDLFLD.f 2. To fix the issue that the last vertical level of CIT in AVIATI10.tm00 of RRFS GTG is marked wrongly, I have a fix on Cactus: /lfs/h2/emc/vpppg/noscrub/yali.mao/git/UPP.v17_hsinmu/sorc/ncep_post.fd/MISCLN.f /lfs/h2/emc/vpppg/noscrub/yali.mao/git/UPP.v17_hsinmu/parm/fv3lam_rrfs.xml /lfs/h2/emc/vpppg/noscrub/yali.mao/git/UPP.v17_hsinmu/parm/postxconfig-NT-fv3lam_rrfs.txt 3. Please add your comments of your code changes into the log history at the beginning of each Fortran code 4. Please merge UPP code updates. Other PRs were merged after your PR was submitted. @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> make sure you pick up above fixes from Yali — Reply to this email directly, view it on GitHub <[#839 (comment)](https://github.com/NOAA-EMC/UPP/pull/839#issuecomment-1908303268)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ALRSTCNBOILQUFIILO536CTYQEORVAVCNFSM6AAAAABAQJVPR2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMBYGMYDGMRWHA> . You are receiving this because you were mentioned.Message ID: ***@***.***>

Great!

WenMeng-NOAA commented:
@hsinmulin-NOAA and @YaliMao-NOAA 
I conducted the UPP standalone tests with 192 tasks on WCOSS2. By adding the new AVIATI dataset, the RRFS runtime increased from  399 seconds to 474 seconds. That performance looks good to me. 
Please verify AVIATI10.tm00 file from my test at /u/wen.meng/ptmp/post_rrfs_2023060614-gtg on Cactus.

hsinmulin-NOAA commented:
The output are identical to the sensitivity test case and all look good to
me.


On Wed, Jan 24, 2024 at 8:02 PM WenMeng-NOAA ***@***.***>
wrote:

> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> and @YaliMao-NOAA
> <https://github.com/YaliMao-NOAA>
> I conducted the UPP standalone tests with 192 tasks on WCOSS2. By adding
> the new AVIATI dataset, the RRFS runtime increased from 399 seconds to 474
> seconds. That performance looks good to me.
> Please verify AVIATI10.tm00 file from my test at
> /u/wen.meng/ptmp/post_rrfs_2023060614-gtg on Cactus.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/UPP/pull/839#issuecomment-1909174526>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCNN6YRNIJBBIK7AK53YQGVILAVCNFSM6AAAAABAQJVPR2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMBZGE3TINJSGY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

WenMeng-NOAA commented:
@hsinmulin-NOAA Please sync your branch with the latest UPP develop? We will process your PR soon. Thanks!

hsinmulin-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

hsinmulin-NOAA commented:
The branch code has been synced and has been compared with the previous
version.  The change is expected
It is done.

On Thu, Jan 25, 2024 at 12:23 PM WenMeng-NOAA ***@***.***>
wrote:

> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> Please sync your
> branch with the latest UPP develop? We will process your PR soon. Thanks!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/UPP/pull/839#issuecomment-1910663244>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCOQ3DHC77ZZ3Q3YDN3YQKIKVAVCNFSM6AAAAABAQJVPR2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMJQGY3DGMRUGQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

WenMeng-NOAA commented:
@FernandoAndrade-NOAA You may start the UPP RTs on Hera and Orion.

WenMeng-NOAA commented:
The UPP RTs were completed on WCOSS2 without changed results.

FernandoAndrade-NOAA commented:
Hera UPP RTs completed successfully with no changes to results. Orion also shows no changes in most tests, however there was an issue between the test script and the gfs_pe test job submission causing it to end early while it sat in queue. I'm rerunning just the gfs test to be sure and will update shortly.

FernandoAndrade-NOAA commented:
> The UPP RTs were completed on WCOSS2 without changed results.

Orion RTs have completed as well. No changes in results, looks good from my side!

WenMeng-NOAA commented:
This PR is ready for merging.
